lib/rubocop/ast/node.rb

```diff
@@ -508,6 +508,21 @@ def guard_clause?
          (block (send (const nil? {:Class :Module}) :new ...) ...)}
       PATTERN
 
+      def_node_matcher :struct_constructor?, <<~PATTERN
+        (block (send (const {nil? (cbase)} :Struct) :new ...) ...)
+      PATTERN
+
+      def_node_matcher :class_definition?, <<~PATTERN
+        {(class (const _ _) ...)
+         (casgn nil? _ (block (send (const {nil? (cbase)} {:Struct :Class}) :new ...) ...))
+         (sclass {(self) (send nil? _)} ...)}
+      PATTERN
+
+      def_node_matcher :module_definition?, <<~PATTERN
+        {(module (const _ _) ...)
+         (casgn nil? _ (block (send (const {nil? (cbase)} :Module) :new ...) ...))}
```

Comment:
The `casgn` is wrong here... Too precise (excluding `Foo::Bar = Module.new do`) and not actually relevant. `include(Module.new do ....` would be strange but should be included.
Be more lazy: be as non-specific as you can 😄